cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -1930,29 +1930,23 @@ TEST(Cast, DictTypeToAnotherDict) {
   check_cast(dictionary(int32(), utf8()), dictionary(int8(), utf8()),
              R"(["a", "b", "a", null])");
 
+  // check float types (NOTE: ArrayFromJSON doesnt work for float value dictionary types)
```

Comment:
What was the error here? Would be nice to file a JIRA for it

Reply:
Done!

Reply:
The error is from `ArrayFromJSON(dictionary(..., floatXX()), ...)` is:
```
NotImplemented: JSON conversion to dictionary<values=float, indices=int8, ordered=0> not implemented
```
Nevertheless, you can `Cast` successfully to `float` values with
```
auto arr = ArrayFromJSON(dictionary(int8(), int32()), "[1, 2, 3, 1, null, 3]");
ASSERT_OK_AND_ASSIGN(auto casted, Cast(arr, dictionary(int8(), float32()), CastOptions::Safe()));
```

Reply:
@nirandaperera I submitted [this PR](https://github.com/apache/arrow/pull/10746) that enables using `ArrayFromJSON` for dictionaries with floating-point values. If PR passes and gets merged, you can change test accordingly and remove error comment.